### PR TITLE
CP-12169: Add common cache to xapi/exauth for PBIS CLI output

### PR DESCRIPTION
**Pull request by phusl**
During AD auth login to xenserver, some pbis command will be called twice(e.g. find-by-sid/list-groups-for-user)
To improve its performance, we need add common cache for the pbis_common function in extauth_plugin_ADpbis.ml

Signed-off-by: Phus Lu <phus.lu@citrix.com>

### CODE REVIEW

**Comment by johnelse** [comment]
@phuslu good idea!

While you're touching this code, I recommend protecting any read or write of `cache_of_pbis_common` with a mutex - for example see how `Threadext.Mutex.execute` is used in [this file](https://github.com/xapi-project/xen-api/blob/3dbdd52d522249e4e5b222e5d02466cc875a4279/ocaml/xapi/pool_role.ml).

**Comment by phusl** [comment]
@johnelse Thank you for the advice!

Since the executing time of "pbis_common" is a bit long,
so I use Threadext.Mutex.execute to guard the whole code block.

File: ocaml/auth/extauth_plugin_ADpbis.ml [hunk 1]
```diff
@@ -669,7 +676,7 @@ let on_enable config_params =
                 Db.Host.set_external_auth_configuration ~__context ~self:host ~value:extauthconf;
                 debug "added external_auth_configuration for host %s" (Db.Host.get_name_label ~__context ~self:host)
             );
-        group_sids_cache := [];
+        cache_of_pbis_common := [];
```
Comment (johnelse) [inline ocaml/auth/extauth_plugin_ADpbis.ml]: This needs a `Mutex.execute` around it too.

**Comment by johnelse** [comment]
Looks good, thanks!

**Pull request merged**